src/GitHub.Exports/Models/LocalRepositoryModel.cs

```diff
@@ -60,12 +61,12 @@ public void Refresh()
         /// <param name="startLine">A specific line, or (if specifying the <paramref name="endLine"/> as well) the start of a range</param>
         /// <param name="endLine">The end of a line range on the specified file.</param>
         /// <returns>An UriString with the generated url, or null if the repository has no remote server configured or if it can't be found locally</returns>
-        public UriString GenerateUrl(string path = null, int startLine = -1, int endLine = -1)
+        public async Task<UriString> GenerateUrl(string path = null, int startLine = -1, int endLine = -1)
         {
             if (CloneUrl == null)
                 return null;
 
-            var sha = HeadSha;
+            var sha = await GitService.GitServiceHelper.GetLatestPushedSha(path);
```

Comment:
Is this the path where the repository is? The `GenerateUrl` method takes a `path` argument that represents the file to generate the url for, not the path to the repository where the file is.